src/api/metadataApi.ts

```diff
@@ -6,11 +6,48 @@
  */
 
 import explorerServiceAxios from './explorerServiceAxios';
+import helpers from '../utils/helpers';
+import { METADATA_RETRY_LIMIT, DOWNLOAD_METADATA_RETRY_INTERVAL } from '../constants';
 
 const metadataApi = {
-  async getDagMetadata(id: string, network: string) {
-    const axios = await explorerServiceAxios(network);
-    return axios.get(`metadata/dag`, { params: { id }});
+  async getDagMetadata(id: string, network: string, options = {}) {
+    type optionsType = {
+      retries: number,
+      retryInterval: number,
+    };
+    const newOptions: optionsType = Object.assign({
+      retries: METADATA_RETRY_LIMIT,
+      retryInterval: DOWNLOAD_METADATA_RETRY_INTERVAL,
+    }, options);
+
+    let { retries, retryInterval } = newOptions;
+    while (retries >= 0) {
+      const axios = await explorerServiceAxios(network);
+      try {
+        const response = await axios.get(`metadata/dag`, { params: { id }});
+        if (response.data) {
+          return response.data;
+        } else {
+          // Error downloading metadata
+          // throw Error and the catch will handle it
+          throw Error;
+        }
+      } catch (e) {
+        if (e.response && e.response.status === 404) {
+          // No need to do anything, the metadata for this token was not found
+          // There is no error here, we just return null
+          return null;
+        } else {
+          // Error downloading metadata, then we should wait a few seconds and retry if still didn't reached retry limit
```

Comment:
I suggest changing the position of this comment to the line just above

`await helpers.sleep(retryInterval);`

And add another comment explaining that when `retries === 0` the method will propagate the error

Reply:
Improved the comments here: https://github.com/HathorNetwork/hathor-wallet-lib/pull/266/commits/0085f4c46108282fde98a7bcb9e536d48d0a8e2d

What do you think?